src/ray/raylet/node_manager.cc

```diff
@@ -360,8 +373,28 @@ void NodeManager::ClientAdded(const ClientTableData &client_data) {
   RAY_LOG(DEBUG) << "[ClientAdded] Received callback from client id " << client_id;
   if (client_id == gcs_client_->client_table().GetLocalClientId()) {
     // We got a notification for ourselves, so we are connected to the GCS now.
-    // Save this NodeManager's resource information in the cluster resource map.
-    cluster_resource_map_[client_id] = initial_config_.resource_config;
+
+    // Delay execution to make sure,
+    // 1) Client cache is up to date.
+    // 2) Lookup resource table after subscribe.
+    io_service_.post([this]() {
```

Comment:
how to guarantee that this lambda will run after the client cache is up to date?

Reply:
The execution of `RequestNotifications` against the client table will push all entries to raylet. Raylet will handle a single notification with multiple client entries and add alive and dead clients to the client cache. So here we lookup resources asynchronously to ensure all client entries are added to the client cache before the resource lookup operation.

Reply:
This change is confusing. I updated the code to lookup resources for only one client at the end of `ClientAdded`.